Debian 12(bookworm), gcc-12, atomic and memory_order

Trying to compile on a system running bookworm with the default g++ (g++ (Raspbian 12.2.0-14+rpi1) 12.2.0).  I get the error
In file included from /usr/include/c++/12/atomic:41,
                 from /home/cdd/projects/LCCV/include/lccv.hpp:5,
                 from /home/cdd/projects/LCCV/src/lccv.cpp:1:
In member function ‘std::__atomic_base<_IntTp>::__int_type std::__atomic_base<_IntTp>::load(std::memory_order) const [with _ITp = bool]’,
    inlined from ‘bool std::atomic<bool>::load(std::memory_order) const’ at /usr/include/c++/12/atomic:112:26,
    inlined from ‘bool lccv::PiCamera::startVideo()’ at /home/cdd/projects/LCCV/src/lccv.cpp:102:20:
/usr/include/c++/12/bits/atomic_base.h:488:31: error: invalid memory model ‘memory_order_release’ for ‘unsigned char __atomic_load_1(const volatile void*, int)’ [-Werror=invalid-memory-model]
  488 |         return __atomic_load_n(&_M_i, int(__m));
      |                ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~

From what I've read "memory_order" should be one of  std::memory_order_relaxed, std::memory_order_consume, std::memory_order_acquire and std::memory_order_seq_cst in an atomic::load.

I've changed line 102 of lccv.cpp from 
  if(running.load(std::memory_order_release)){
to 
  if(running.load(std::memory_order_consume)){
lccv  now compiles but I have no idea if std::memory_order_release is the correct value to use.

--- a/src/lccv.cpp
+++ b/src/lccv.cpp
@@ -99,7 +99,7 @@ bool PiCamera::capturePhoto(cv::Mat &frame)
 bool PiCamera::startVideo()
 {
     if(camerastarted)stopPhoto();
-    if(running.load(std::memory_order_release)){
+    if(running.load(std::memory_order_relaxed)){
         std::cerr<<"Video thread already running";
         return false;
     }

--- a/src/libcamera_app.cpp
+++ b/src/libcamera_app.cpp
@@ -307,7 +307,6 @@ void LibcameraApp::queueRequest(CompletedRequest *completed_request)
 	BufferMap buffers(std::move(completed_request->buffers));
 
 	Request *request = completed_request->request;
-	delete completed_request;
 	assert(request);
 
 	// This function may run asynchronously so needs protection from the
@@ -321,6 +320,7 @@ void LibcameraApp::queueRequest(CompletedRequest *completed_request)
 	{
 		std::lock_guard<std::mutex> lock(completed_requests_mutex_);
 		auto it = completed_requests_.find(completed_request);
+        delete completed_request;
 		if (it == completed_requests_.end())
 			return;
 		completed_requests_.erase(it);

